easybuild/easyconfigs/j/jax/jax-0.2.18-foss-2020b.eb

```diff
@@ -0,0 +1,35 @@
+easyblock = 'PythonPackage'
+
+name = 'jax'
+version = '0.2.18'
+
+homepage = 'https://pypi.python.org/pypi/jax'
+description = """Composable transformations of Python+NumPy programs: differentiate, vectorize, JIT to GPU/TPU, and
+more"""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+
+# source_urls = [PYPI_SOURCE]
+# sources = [SOURCE_TAR_GZ]
+
+source_urls = ['https://github.com/google/jax/archive/']
+sources = ['%(name)s-v%(version)s.tar.gz']
+patches = [
+    'jax-0.2.18-correct_libraries.patch',
+]
+
+preinstallopts = 'python3 build/build.py --output_user_root=%(builddir)s --bazel_path="$EBROOTBAZEL/bin/bazel" && '
+preinstallopts += 'pip install dist/*.whl && '
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('Bazel', '3.7.2'),
+    ('SciPy-bundle', '2020.11'),
+    ('Cython', '0.29.22'),
+    # ('TensorFlow', '2.5.0'), do we need this?
```

Comment:
Based on the feedback in https://github.com/google/jax/discussions/7467, including TensorFlow as a dependency doesn't make sense.

When building `jax` from source, I think we should try and control which TensorFlow sources are used, by pre-downloading them and making sure that Bazel picks them up (rather than downloading stuff itself).

It seems like that's doable by:

* adding the TensorFlow 2.5.0 tarball as an additional source
* patching `jax/WORKSPACE` so it uses that rather than the TensorFlow commit that is specified now; I guess you could try to replace the `https://` URL with a `file://` URL?

You'll need to do the latter at runtime (with `sed`, not a patch file), since the location where the pre-downloaded TensorFlow source tarball is located is not known up front.

Through @Flamefire ?

Reply:
I guess you can basically use the local_repository, see https://github.com/easybuilders/easybuild-easyconfigs/pull/13622/files#r683278973 for that.
This would at least allow to use a patch for the TF sources as required (the default_shell_env stuff) and only need the sed for the path (the patch can insert a nice marker where the sed can apply)

Reply:
done